dbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalSegment.cpp

```diff
@@ -961,16 +964,23 @@ bool DeltaMergeStore::segmentWaitDeltaLocalIndexReady(const SegmentPtr & segment
         if (!column_file_persisted_set)
             return false; // ColumnFilePersistedSet is not exist, return false
         bool all_indexes_built = true;
-        for (const auto & index : *build_info.indexes_to_build)
+        if (auto lock = delta_weak_ptr.lock()->getLock(); lock)
```

Comment:
If `delta_weak_ptr.lock()` return nullptr, we'd better let it throw exception or let it return false to skip the task rather than just crash